Added network policy trouble shooting guide

olaf-meyer opened the pull request:
If you add a project to a control plan, a network policy is added to the new data plane project which allows access pods with the label `maistra.io/expose-route: 'true'` only from namespace with the label `network.openshift.io/policy-group: ingress`. This causes that routes to the data plane no longer work, if the pods have not been label. To avoid problem I updated the documentation.

You can find the commit in the operator here: https://github.com/Maistra/istio-operator/commit/bf3d7df598cb2cbe5d5c7ca43b52eda1190254e9

olaf-meyer committed:
Added network policy trouble shooting guide

brian-avery commented on topics/docs/troubleshooting/troubleshooting_network_policy.adoc:
```diff
@@ -0,0 +1,19 @@
+=== Routes Not Working
+If you add a project to the data plane, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. Since adding a project to a data plane, a netwok rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.
+
+For deployments you add the label `maistra.io/expose-route: 'true' `to the pods with the following command:
```
Thanks for your contribution. 

Small comment, this looks like it didn't render properly:

![image](https://user-images.githubusercontent.com/8484260/83528864-9db3d400-a4b7-11ea-8df1-1be9e67ac4ce.png)

olaf-meyer replied:
Hello Brian,

Sorry I missed that point. I have to admit that I just used the preview in my code editor, because I didn't want to install hugo and the other components. Also, I noticed that the link to the troubleshooting page was incorrect. I fixed both errors. 

After your feedback I created for me a container which allows me to verify my changes locally. Should I create a pull request for this, so that other users can use this container as well?

Kind regards,

Olaf

brian-avery replied:
We have a build container that we use which has all of the tooling in it already. I can look into combining that with our makefile in order to allow the commands to all be run through docker without the installation of additional tooling. I've created [Maistra-1543](https://issues.redhat.com/browse/MAISTRA-1543) to track it.

brian-avery replied:
This is our image: https://quay.io/repository/maistra-dev/maistra-builder?tab=tags. We have tags for Maistra 1.1 and Maistra 1.2.

olaf-meyer committed:
Fixed typo and changed link to troubleshooting page

dgn commented:
/ok-to-test

olaf-meyer committed:
Fixed typos

olaf-meyer commented:
Hello,

I created a new file with this pull request and the check is testing, if this file already exist in the branch. Is there a way to overwrite this failed check?

Kind regards,

Olaf

brian-avery commented:
Hey @olaf-meyer 

I think I know what's going on here. Our documentation generates edit links on build such that someone can just click the edit button on the page they are viewing and jump directly to the file on GitHub to fix it. If this is a new file (which in this case it is), then the link checker will see the auto-generated link and try to verify that it exists. Because it's being created by the pull request, the link won't exist yet. 

I think we have two ways of fixing this: 
* ignore any edit links
* we should be able to get a list of files which changed in this PR. We can likely use this to ignore the edit link for the specific file that has changed. 

In terms of this PR, I don't think there's anything broken, just the bot misbehaving. I'll try to get a fix in for this tomorrow.

brian-avery commented:
/retest

dgn commented:
/retest

dgn commented:
/retest

dgn commented:
/retest

dgn commented on topics/docs/troubleshooting/troubleshooting_network_policy.adoc:
```diff
@@ -0,0 +1,19 @@
+oc patch deployment <deployment name> -p '{"spec":{"template":{"metadata":{"labels":{"maistra.io/expose-route":"true"}}}}}'
+----
+
+In certain cases i.e run the OpenShift cluster on VMWare, you might need to add the label `network.openshift.io/policy-group: ingress` to the default namespace. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html
```
```suggestion
In certain cases (e.g. when running the OpenShift cluster on VMWare), your ingress controller is running on the host network. In this case, ingress traffic will appear to originate in the `default` namespace. In this case you might want to add the label `network.openshift.io/policy-group: ingress` to the default namespace. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html
```

brian-avery replied:
```suggestion
In certain cases i.e run the OpenShift cluster on VMWare, you might need to add the label `network.openshift.io/policy-group: ingress` to the default project. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html
```

brian-avery committed:
Merge branch 'maistra-1.2' into maistra-1.2

dgn commented on topics/docs/troubleshooting/troubleshooting_network_policy.adoc:
```diff
@@ -0,0 +1,19 @@
+
+
+=== Routes Not Working
+If you add a project to the data plane, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. Since adding a project to a data plane, a network rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.
```
```suggestion
If you add a project to the mesh, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. When adding a project to a mesh, a network rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.
```

dgn commented on topics/docs/troubleshooting/troubleshooting_network_policy.adoc:
```diff
@@ -0,0 +1,19 @@
+=== Routes Not Working
+If you add a project to the data plane, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. Since adding a project to a data plane, a network rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.
+
+For deployments you add the label `maistra.io/expose-route: 'true'` to the pods with the following command:
```
```suggestion
If you're using Deployments, you can add the label `maistra.io/expose-route: 'true'` to the pods with the following command:
```

dgn reviewed (commented):
Thanks for the contribution! I left some comments re: wording

brian-avery commented on vocab.txt:
```diff
@@ -17,6 +17,7 @@ Jira
 Maistra
 multitenant
 Multitenancy
+namespace
```
If you use project above, namespace shouldn't need to be added.

olaf-meyer committed:
Update topics/docs/troubleshooting/troubleshooting_network_policy.adoc

Good point. The description is much clearer.

Co-authored-by: Daniel Grimm <dgrimm@redhat.com>

olaf-meyer committed:
Update topics/docs/troubleshooting/troubleshooting_network_policy.adoc

Good point. The description is much clearer when using mesh instead of data plan.

Co-authored-by: Daniel Grimm <dgrimm@redhat.com>

olaf-meyer committed:
Update topics/docs/troubleshooting/troubleshooting_network_policy.adoc

The description is much clearer.

Co-authored-by: Daniel Grimm <dgrimm@redhat.com>

olaf-meyer committed:
Removed word namespace, because it is no longer required.

olaf-meyer committed:
Replaced namespace with project

olaf-meyer commented:
Thank you very much for your patience. The suggestions are much better than my text.

dgn reviewed (approved):
Thank you!

olaf-meyer committed:
Added feature to run maistra documentation webserver in a docker container

olaf-meyer committed:
Update Makefile

olaf-meyer committed:
Delete Dockerfile

brian-avery commented on topics/docs/installation/configuring-members.adoc:
```diff
@@ -23,6 +23,10 @@ and must be named `default`. An example resource can be seen below:
 {{< snippet_markdown name="istio_installation_smmr" >}}
 
+{{% notice tip %}}
+There might be configurations where routes to pods are no longer working after adding a project to a control plan. Please see the link:../../troubleshooting/troubleshooting_network_policy[network policy trouble shooting guide] to solve the problem.
```
```suggestion
There might be configurations where routes to pods are no longer working after adding a project to a control plane. Please see the link:../../troubleshooting/troubleshooting_network_policy[network policy trouble shooting guide] to solve the problem.
```

brian-avery replied:
Thanks again for this. One minor spelling error and I'll approve :)

olaf-meyer committed:
Update topics/docs/installation/configuring-members.adoc

Co-authored-by: Brian Avery <bavery@redhat.com>
